### PR TITLE
Made cooldown server-side

### PR DESCRIPTION
**Pull request by TheNickSkater**
The cooldown is now only set on the Server and synced to the client. This will fix many issues with clientside cooldown not working correctly.

I also already have fixed the classes in the heroes pack. I will make a PR if this gets approved.

If you are unsure you can of course test this. But I have it running like this on my server for about a month now without any issues, it even works better.

### CODE REVIEW

File: lua/classes/shared/sh_player.lua [hunk 1]
```diff
@@ -272,17 +270,26 @@ function plymeta:ClassDeactivate()
...
 
-	if classData.OnAbilityDeactivate and isfunction(classData.OnAbilityDeactivate) then
-		cooldown = not classData.OnAbilityDeactivate(self)
+		local cooldown = true
+
+		if classData.OnAbilityDeactivate and isfunction(classData.OnAbilityDeactivate) then
+			cooldown = not classData.OnAbilityDeactivate(self)
+		end
+
+		self.classTimestamp = nil -- Still used???
```
Comment (saibotk) [inline lua/classes/shared/sh_player.lua:280]: Can be removed, as I could not find any reference to this at all, but then please also remove it from `lua/classes/shared/sh_player.lua`

Comment (TheNickSkater) [reply lua/classes/shared/sh_player.lua:280]: The classTimestamp is actually still used? For example in the abilitymeter.

Comment (saibotk) [reply lua/classes/shared/sh_player.lua:280]: Where?

Comment (saibotk) [reply lua/classes/shared/sh_player.lua:280]: Ah nevermind

**Comment by TheNickSkater** [comment]
Requested changes updated.

File: lua/classes/shared/sh_player.lua [hunk 2]
```diff
@@ -272,17 +270,26 @@ function plymeta:ClassDeactivate()
...
+
+		if cooldown and classData.cooldown ~= 0 then
+			self:SetClassCooldown(classData.cooldown)
+			self:SetClassCooldownTS(CurTime())
+			self:SyncClassState()
+		end
 	end
 
-	self.classTimestamp = nil -- Still used???
+	if SERVER then return end
```
Comment (saibotk) [inline lua/classes/shared/sh_player.lua:289]: Move this return two lines up into the existing `if SERVER` block

Comment (TheNickSkater) [reply lua/classes/shared/sh_player.lua]: Moved

File: lua/classes/shared/sh_player.lua (hunk 1 above)
Comment (saibotk) [reply lua/classes/shared/sh_player.lua:280]: Remove the `-- Still used???` comment

**Pull request merged**